modules/rofi/module.nix

```diff
@@ -0,0 +1,176 @@
+
+    mkKeyValue =
+      {
+        sep ? ": ",
+        end ? ";",
+      }:
+      name: value: "${name}${sep}${mkValueString value}${end}";
+
+    mkRasiSection =
+      name: value:
+      if isAttrs value then
+        let
+          toRasiKeyValue = lib.generators.toKeyValue { mkKeyValue = mkKeyValue { }; };
+          # Remove null values so the resulting config does not have empty lines
+          configStr = toRasiKeyValue (filterAttrs (_: v: v != null) value);
+        in
+        ''
+          ${name} {
+          ${configStr}}
+        ''
+      else
+        (mkKeyValue {
+          sep = " ";
+          end = "";
+        } name value)
+        + "\n";
+
+    toRasi =
+      attrs:
+      lib.concatStringsSep "\n" (
+        lib.concatMap (lib.mapAttrsToList mkRasiSection) [
+          (filterAttrs (n: _: n == "@theme") attrs)
+          (filterAttrs (n: _: n == "@import") attrs)
+          (removeAttrs attrs [
+            "@theme"
+            "@import"
+          ])
+        ]
+      );
+
+    rasiLiteral =
+      types.submodule {
+        options = {
+          _type = lib.mkOption {
+            type = types.enum [ "literal" ];
+            internal = true;
+          };
+
+          value = lib.mkOption {
+            type = types.str;
+            internal = true;
+          };
+        };
+      }
+      // {
+        description = "Rasi literal string";
+      };
+
+    primitive =
+      with types;
+      (oneOf [
+        str
+        int
+        bool
+        rasiLiteral
+      ]);
+
+    # Either a `section { foo: "bar"; }` or a `@import/@theme "some-text"`
+    configType = with types; (either (attrsOf (either primitive (listOf primitive))) str);
+
+    themeType = with types; attrsOf configType;
+
+    theme =
+      if (isAttrs config.theme) then
+        builtins.toPath (
+          config.pkgs.writeTextFile {
+            name = "rofi-theme";
+            text = toRasi config.theme;
+          }
+        )
+      else
+        config.theme;
+  in
+  {
+    options = {
+      settings = lib.mkOption {
+        type = configType;
+        default = {
+          location = 0;
+          xoffset = 0;
+          yoffset = 0;
+        };
+        description = "Configuration settings for rofi.";
+      };
+      plugins = lib.mkOption {
+        type = lib.types.listOf lib.types.package;
+        default = [ ];
+        description = "List of rofi plugins to be installed";
+      };
+      theme = lib.mkOption {
+        type =
+          with lib.types;
+          nullOr (oneOf [
+            str
+            path
+            themeType
+          ]);
+        default = null;
+      };
+
+      extraFlags = lib.mkOption {
+        type = lib.types.attrsOf lib.types.unspecified; # TODO add list handling
+        default = { };
+        description = "Extra flags to pass to rofi.";
+      };
+    };
+
+    config.flags = {
+      "-config" = config.pkgs.writeTextFile {
```

Comment:
it would be nice if this textfile could be generated through a wlib.types.file option. so people could override the path if they want to have a config file outside the store (or read from an environment file)